src/Application/ActiveFileProcessorsProvider.php

```diff
@@ -0,0 +1,35 @@
+<?php
+declare(strict_types=1);
+
+namespace Rector\Core\Application;
+
+use Rector\Core\Contract\Processor\FileProcessorInterface;
+use Symplify\Skipper\Skipper\Skipper;
+
+final class ActiveFileProcessorsProvider
+{
+    /**
+     * @var FileProcessorInterface[]
+     */
+    private $fileProcessors = [];
+
+    /**
+     * @param FileProcessorInterface[] $fileProcessors
+     */
+    public function __construct(Skipper $skipper, array $fileProcessors)
+    {
+        $this->fileProcessors = array_filter($fileProcessors, function (FileProcessorInterface $fileProcessor) use (
+            $skipper
+        ) {
+            return ! $skipper->shouldSkipElement($fileProcessor);
+        });
+    }
+
```

Comment:
This should be done on rule-level. 
The same way all PHP rules are not skipped not, just specific ones.

Reply:
But, how could someone just ignore a complete FileProcessor which is part a set? In my case this would make sense, but i agree, it is different from current approach.